server/src/settings/Settings.ts

```diff
@@ -0,0 +1,62 @@
+/** Describe the YML relevant settings part as defined in `package.json`. */
+export interface IYseopmlSettings {
+    yseopml: IYseopmlServerSettings;
+}
+
+/** Describe the _server_ relevant settings part as defined in `package.json`. */
+export interface IYseopmlServerSettings {
+    activateParsingProblemsReporting: boolean;
+    parseAllProjectFilesAtStartup: boolean;
+    pathToPredefinedObjectsXml: string;
+    ymlParsingIssueSeverityLevel: string;
+    kaoFiles: string[];
+    documentFormat: IDocumentFormatSettings;
+}
+
+/**
+ * Possible attributes that can be sent as document format settings as defined in `package.json`.
+ */
+export interface IDocumentFormatSettings {
+    // 'yes' and 'no' acts like a new type. Impossible to use const.
+    /** `yes` to enable the document format feature. `no` otherwise. */
+    enableDocumentFormat: 'yes' | 'no';
+    /** `true` to add semicolon when possible. `false` otherwise. */
+    semicolonWhenPossible: boolean;
+    /**
+     * `true` to set a one space distance between some keywords and parenthesis. \
+     * `false` to have this distance set to zero.
+     */
+    spaceBetweenKeywordAndParenthesis: boolean;
+    // More options to come.
+}
+
+/**
+ * Default values for the document format settings, as defined in `package.json`.
+ */
+export const DEFAULT_DOC_FORMAT_SETTINGS: IDocumentFormatSettings = {
+    enableDocumentFormat: 'yes',
+    semicolonWhenPossible: true,
+    spaceBetweenKeywordAndParenthesis: false,
+};
+
+/**
+ * Set in an instance of document format settings set the default values for every undefined options.
+ *
+ * @param documentFormat a document format settings set
+ *
+ * @return the provided instance with default values for every attribute that was undefined
+ * or `DEFAULT_DOC_FORMAT_SETTINGS` if the instance was falsy.
+ */
+export function setDocumentFormatDefaultValues(documentFormat: IDocumentFormatSettings): IDocumentFormatSettings {
+    if (!documentFormat) {
+        return DEFAULT_DOC_FORMAT_SETTINGS;
+    }
+    documentFormat.enableDocumentFormat =
+        documentFormat.enableDocumentFormat ?? DEFAULT_DOC_FORMAT_SETTINGS.enableDocumentFormat;
+    documentFormat.semicolonWhenPossible =
+        documentFormat.semicolonWhenPossible ?? DEFAULT_DOC_FORMAT_SETTINGS.semicolonWhenPossible;
+    documentFormat.spaceBetweenKeywordAndParenthesis =
+        documentFormat.spaceBetweenKeywordAndParenthesis ??
+        DEFAULT_DOC_FORMAT_SETTINGS.spaceBetweenKeywordAndParenthesis;
```

Comment:
* Maybe you could do something similar to my…
https://github.com/PrismJS/prism/blob/master/components/prism-bash.js#L175-L195
… in order to just loop through properties (maybe even getting the list of relevant properties automatically, with no need for further maintenance when adding setting options)

* Huuuuuuuum won’t the `??` screw up if the option has a boolean type and was set to `false` by the user? If default=true and user-choice=false, won’t the user choice be overridden? Maybe I’m wrong.

Reply:
> Maybe you could do something similar to my…
https://github.com/PrismJS/prism/blob/master/components/prism-bash.js#L175-L195
… in order to just loop through properties (maybe even getting the list of relevant properties automatically, with no need for further maintenance when adding setting options)

Maybe there is a better way to do so. https://attacomsian.com/blog/javascript-iterate-objects#:~:text=keys()%20Method-,The%20Object.,the%20value%20of%20each%20property.

> Huuuuuuuum won’t the `??` screw up if the option has a boolean type and was set to `false` by the user? If default=true and user-choice=false, won’t the user choice be overridden? Maybe I’m wrong.


`let x = foo ?? bar();` is equivalent to `let x = foo !== null && foo !== undefined ? foo : bar();`  (see [Typescript reference website](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing))